File: onebot-sdk/src/main/java/cn/evole/onebot/sdk/event/notice/misc/ReceiveOfflineFilesNoticeEvent.java
package cn.evole.onebot.sdk.event.notice.misc;

import cn.evole.onebot.sdk.event.notice.NoticeEvent;
import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

/**
 * Created on 2022/7/8.
 *
 * @author cnlimiter
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder(toBuilder = true)
@EqualsAndHashCode(callSuper = true)
public class ReceiveOfflineFilesNoticeEvent extends NoticeEvent {

    @SerializedName("file")
    private File file;

    /**
     * 文件对象
     */
    @Data
    public static class File {

        @SerializedName("name")
        private String name;

        @SerializedName("size")
        private long size;

        @SerializedName("url")
        private String url;

    }

}

File: onebot/src/main/kotlin/sdk/event/notice/misc/ReceiveOfflineFilesNoticeEvent.kt
package cn.evolvefield.onebot.sdk.event.notice.misc

import cn.evolvefield.onebot.sdk.event.notice.NoticeEvent
import com.google.gson.annotations.SerializedName

class ReceiveOfflineFilesNoticeEvent : NoticeEvent() {
    @SerializedName("file")
    var file: File? = null

    /**
     * 文件对象
     */
    class File {
        @SerializedName("name")
        var name = ""
        @SerializedName("size")
        var size = 0L
        @SerializedName("url")
        var url = ""
    }
}
